### DISCUSSION TITLE
libsodium requirement issue

### DISCUSSION THREAD

**Discussion by sarfarazsavvy** [lcobucci/jwt#784] [Q&A] [answered]
Hello Community,
we are facing issue, this package requires libsodium previously ext-libsodium was available to install in php 7.2 but in php 7.4 libsodium is not installing. is there any way to ignore libsodium is there any alternative way.

**Comment by SvenRtbg** [comment]
Libsodium is supposed to be bundled with PHP since 7.2, so you have to fix the issue with whatever your PHP provider has done to make it a problem.

The extension provides modern cryptographic functions and is essential to this library.
-- 
Viele Grüße
Sven Rautenberg

**Comment by lcobucci** [comment, marked as answer]
@sarfarazsavvy you can use composer's `--ignore-platform-req=ext-sodium` (https://getcomposer.org/doc/03-cli.md#install-i) and check if you can run things but you'd essentially be on your own.

The recommended approach is to have the extension installed (that depends on the OS you have).

**Comment by donatj** [comment]
@lcobucci - does https://github.com/lcobucci/jwt/blob/4.2.x/src/SodiumBase64Polyfill.php not polyfill sodium? Is it necessary as a requirement?

**Reply by lcobucci** [reply]
The polyfill was introduced because of things we can't require via composer (https://github.com/lcobucci/jwt/issues/665).

The plan is to eventually be able to remove the polyfill and only rely on the extension.
